imap_processing/lo/l1a/lo_l1a_write_cdfs.py

```diff
@@ -0,0 +1,123 @@
+        for one pointing.
+    """
+    created_filepaths = []
+
+    # Write Science Direct Events CDF if available
+    science_direct_events = data.filter_apid(LoAPID.ILO_SCI_DE.value)
+    if science_direct_events:
+        scide_dataset = create_lo_scide_dataset(science_direct_events)
+        cdf_file = write_cdf(scide_dataset)
+        created_filepaths.append(cdf_file)
+
+    # TODO: Add the rest of the APIDS
+
+    return created_filepaths
+
+
+def create_lo_scide_dataset(scide: list):
+    """
+    Create Lo L1A Science Direct Event Dataset from the ScienceDirectEvent dataclasses.
+
+    Parameters
+    ----------
+    scide: list
+        List of ScienceDirectEvent data classes.
+
+    Returns
+    -------
+    xarray.Dataset
+        Lo L1A Science Direct Event Dataset.
+    """
+    # TODO: getting sci_de_times because it's used in both the data time field
+    # and epoch. Need to figure out if this is needed and if a conversion needs
+    # to happen to get the epoch time.
+    sci_de_times = np.concatenate([sci_de_data.TIME for sci_de_data in scide])
+    sci_de_time = xr.DataArray(
+        sci_de_times, dims="epoch", attrs=lo_cdf_attrs.lo_tof_attrs.output()
+    )
+    sci_de_epoch = xr.DataArray(
+        np.array(sci_de_times, dtype="datetime64[s]"),
+        dims=["epoch"],
+        name="epoch",
+        attrs=ConstantCoordinates.EPOCH,
+    )
+    sci_de_energy = xr.DataArray(
+        np.concatenate([sci_de_data.ENERGY for sci_de_data in scide]),
+        dims="epoch",
+        attrs=lo_cdf_attrs.lo_tof_attrs.output(),
+    )
+    sci_de_mode = xr.DataArray(
+        np.concatenate([sci_de_data.MODE for sci_de_data in scide]),
+        dims="epoch",
+        attrs=lo_cdf_attrs.lo_tof_attrs.output(),
+    )
+    sci_de_tof0 = xr.DataArray(
+        np.concatenate([sci_de_data.TOF0 for sci_de_data in scide]),
+        dims="epoch",
+        attrs=lo_cdf_attrs.lo_tof_attrs.output(),
+    )
+    sci_de_tof1 = xr.DataArray(
+        np.concatenate([sci_de_data.TOF1 for sci_de_data in scide]),
+        dims="epoch",
+        attrs=lo_cdf_attrs.lo_tof_attrs.output(),
+    )
+    sci_de_tof2 = xr.DataArray(
+        np.concatenate([sci_de_data.TOF2 for sci_de_data in scide]),
+        dims="epoch",
+        attrs=lo_cdf_attrs.lo_tof_attrs.output(),
+    )
+    sci_de_tof3 = xr.DataArray(
+        np.concatenate([sci_de_data.TOF3 for sci_de_data in scide]),
+        dims="epoch",
+        attrs=lo_cdf_attrs.lo_tof_attrs.output(),
+    )
+    sci_de_checksum = xr.DataArray(
+        np.concatenate([sci_de_data.CKSM for sci_de_data in scide]),
+        dims="epoch",
+        attrs=lo_cdf_attrs.lo_tof_attrs.output(),
+    )
+    sci_de_pos = xr.DataArray(
+        np.concatenate([sci_de_data.POS for sci_de_data in scide]),
+        dims="epoch",
+        attrs=lo_cdf_attrs.lo_tof_attrs.output(),
+    )
+
+    # Create the full dataset
+    sci_de_dataset = xr.Dataset(
+        data_vars={
+            "time": sci_de_time,
```

Comment:
can you name this `de_time` instead of `time`? It's more description. On first glance, I thought it was epoch time.

Reply:
Yes, I'll change that field name and the variable names in the direct event class to de_time